lib/web_ui/lib/src/engine/surface/scene_builder.dart

```diff
@@ -157,6 +157,23 @@ class SurfaceSceneBuilder implements ui.SceneBuilder {
     throw UnimplementedError();
   }
 
+  /// Pushes an image filter operation onto the operation stack.
+  ///
+  /// The given filter is applied to the children's rasterization before compositing them into
+  /// the scene.
+  ///
+  /// {@macro dart.ui.sceneBuilder.oldLayer}
+  ///
+  /// {@macro dart.ui.sceneBuilder.oldLayerVsRetained}
+  ///
+  /// See [pop] for details about the operation stack.
+  @override
+  ui.ImageFilterEngineLayer pushImageFilter(ui.ImageFilter filter,
+                                            {ui.ImageFilterEngineLayer oldLayer}) {
```

Comment:
trivial nit: use flutter style:

```
  ui.ImageFilterEngineLayer pushImageFilter(ui.ImageFilter filter, {
    ui.ImageFilterEngineLayer oldLayer,
  }) {
```

Reply:
To be fair, what I used is suggested by the built-in formatter in Android Studio with Dart plugin, but it does look like it conflicts with our guidelines in the Contributing docs. I was actually trying to match the other nearby push...() methods and it was the closest that the A.S. formatter would allow. But even those nearby methods seem to violate the 2-space indentation rule, so neither my new addition, nor the (majority of the) existing push() methods seem to match proper style.

Matching the nearby methods would involve moving my indent back to 4 spaces (should be 2 per the formatting guidelines) but leaving the curly braces in-line with the optional named argument.

The formatter seems to agree with what you are suggesting so that is the path of least editing resistance in the future and I agree that it looks nicer. The question I guess that I have would be which of the following is the best approach when editing in an existing source file that already has an established (if flawed) pattern:

- Try to match what is around the code
- Use proper formatting for the changes even if it conflicts with/stands out from the surrounding code
- Fix all of the code even though it introduces diffs on lines that aren't changing for the current work being done

I usually go for similarity when adding a feature or fixing a bug and try to keep bookkeeping changes like reformatting passes in their own push.

Reply:
dartfmt seems to be the origin of the existing indentations. If I run it on the existing files (before my addition here) then it seems to like them.

Reply:
I submitted https://github.com/flutter/engine/pull/14539 to address code formatting in this file - hopefully to push that before this change so I can fit in to the existing code style rather than pioneer something here...

Reply:
https://github.com/flutter/engine/pull/14539 is progressing through approvals. I will adjust my additions to match that style when it goes in.